imap/search_squat.c

```diff
@@ -93,17 +93,22 @@ typedef struct {
 static const char *squat_strerror(int err);
 
 static const char * const doctypes_by_part[SEARCH_NUM_PARTS] = {
-    "mh",
-    "f",
-    "t",
-    "c",
-    "b",
-    "s",
-    "h",
-    "m",
-    "o",
-    "a",
-    "ab"
+    "msh", // SEARCH_PART_ANY
+    "f",   // SEARCH_PART_FROM
+    "t",   // SEARCH_PART_TO
+    "c",   // SEARCH_PART_CC
+    "b",   // SEARCH_PART_BCC
+    "s",   // SEARCH_PART_SUBJECT
+    NULL,  // SEARCH_PART_LISTID
+    NULL,  // SEARCH_PART_TYPE
+    "h",   // SEARCH_PART_HEADERS
+    "m",   // SEARCH_PART_BODY
+    "o",   // SEARCH_PART_LOCATION
+    "a",   // SEARCH_PART_ATTACHMENTNAME
+    NULL,  // SEARCH_PART_ATTACHMENTBODY
+    NULL,  // SEARCH_PART_DELIVEREDTO
+    NULL,  // SEARCH_PART_LANGUAGE
+    NULL   // SEARCH_PART_PRIORITY
```

Comment:
Not sure if `"mh"` -> `"msh"` is a typo?  Though thinking about it more, I think "msh" means "body or subject or headers", which makes a vague sense.

Reply:
Yes, it means to also search subjects for TEXT search. that's consistent with how we do text search for the Xapian backend.